internal/services/inference/deployment.go

```diff
@@ -0,0 +1,349 @@
+				PrivateNetwork: &inference.EndpointSpecPrivateNetwork{
+					PrivateNetworkID: regional.ExpandID(privateID.(string)).ID,
+				},
+				DisableAuth: privateEndpointMap["disable_auth"].(bool),
+			}
+			endpoints = append(endpoints, &privateEp)
+		}
+	}
+
+	return endpoints
+}
+
+func ResourceDeploymentRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	api, region, id, err := NewAPIWithRegionAndID(m, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	deployment, err := waitForDeployment(ctx, api, region, id, d.Timeout(schema.TimeoutRead))
+	if err != nil {
+		if httperrors.Is404(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	_ = d.Set("name", deployment.Name)
+	_ = d.Set("region", deployment.Region)
+	_ = d.Set("project_id", deployment.ProjectID)
+	_ = d.Set("node_type", deployment.NodeType)
+	_ = d.Set("model_name", deployment.ModelName)
+	_ = d.Set("min_size", int(deployment.MinSize))
+	_ = d.Set("max_size", int(deployment.MaxSize))
+	_ = d.Set("size", int(deployment.Size))
+	_ = d.Set("status", deployment.Status)
+	_ = d.Set("model_id", deployment.ModelID)
+	_ = d.Set("created_at", types.FlattenTime(deployment.CreatedAt))
+	_ = d.Set("updated_at", types.FlattenTime(deployment.UpdatedAt))
+	var privateEndpoints []map[string]interface{}
+	var publicEndpoints []map[string]interface{}
+
+	for _, endpoint := range deployment.Endpoints {
+		if endpoint.PrivateNetwork != nil {
+			privateEndpointSpec := map[string]interface{}{
+				"id":                  endpoint.ID,
+				"private_endpoint_id": regional.NewID(deployment.Region, endpoint.PrivateNetwork.PrivateNetworkID).String(),
+				"disable_auth":        endpoint.DisableAuth,
+				"url":                 endpoint.URL,
+			}
+			privateEndpoints = append(privateEndpoints, privateEndpointSpec)
+		}
+		if endpoint.PublicAccess != nil {
+			publicEndpointSpec := map[string]interface{}{
+				"id":           endpoint.ID,
+				"is_enabled":   true,
+				"disable_auth": endpoint.DisableAuth,
+				"url":          endpoint.URL,
+			}
+			publicEndpoints = append(publicEndpoints, publicEndpointSpec)
+		}
+	}
+
+	if len(privateEndpoints) > 0 {
+		_ = d.Set("private_endpoint", privateEndpoints)
+	}
+	if len(publicEndpoints) > 0 {
+		_ = d.Set("public_endpoint", publicEndpoints)
+	}
+	return nil
+}
+
+func ResourceDeploymentUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	api, region, id, err := NewAPIWithRegionAndID(m, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	deployment, err := waitForDeployment(ctx, api, region, id, d.Timeout(schema.TimeoutUpdate))
+	if err != nil {
+		if httperrors.Is404(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	req := &inference.UpdateDeploymentRequest{
+		Region:       region,
+		DeploymentID: deployment.ID,
+	}
+
+	if d.HasChange("name") {
+		req.Name = types.ExpandUpdatedStringPtr(d.Get("name"))
+	}
```

Comment:
Api allows more field in Update: https://github.com/scaleway/scaleway-sdk-go/blob/a2c1e85de57a916ea17d27c4efe2646fc5f33d11/api/inference/v1beta1/inference_sdk.go#L786-L803